[CI] RestEsqlIT testTopLevelFilterBoolMerged {SYNC} failing

**Build Scans:**
- [elasticsearch-periodic #3711 / release-tests](https://gradle-enterprise.elastic.co/s/t6pceldi7cczm)
- [elasticsearch-periodic #3707 / release-tests](https://gradle-enterprise.elastic.co/s/curg3m6yyrsaw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.RestEsqlIT.testTopLevelFilterBoolMerged {SYNC}" -Dtests.seed=3863D2260DB56A6A -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=th-TH -Dtests.timezone=Europe/Vienna -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testTopLevelFilterBoolMerged%20%7BSYNC%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.qa.multi_node.RestEsqlIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
org.elasticsearch.client.ResponseException: method [POST], host [http://[::1]:44713], URI [/_query?pretty=true&error_trace=true], status line [HTTP/1.1 500 Internal Server Error]
Warnings: [No limit defined, adding default limit of [1000]]
:)
úerrorúroot_causeøútypeHexceptionreason!stack_traceäorg.elasticsearch.ElasticsearchException
	at org.elasticsearch.compute.operator.FailureCollector.unwrapAndCollect(FailureCollector.java:51)
	at org.elasticsearch.xpack.esql.plugin.ComputeListener.lambda$acquireAvoid$2(ComputeListener.java:61)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:187)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:192)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:306)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:409)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:264)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:92)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:48)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleResponse(TransportService.java:1775)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1498)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1595)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel$1.doRun(TransportService.java:1575)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
	Suppressed: org.elasticsearch.tasks.TaskCancelledException: cancelled on failure
		at org.elasticsearch.compute.operator.Driver.cancel(Driver.java:304)
		at org.elasticsearch.compute.operator.DriverTaskRunner$DriverRequest$1.onCancelled(DriverTaskRunner.java:101)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.CancellableTask.cancel(CancellableTask.java:43)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager$CancellableTaskHolder.cancel(TaskManager.java:617)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager.cancel(TaskManager.java:274)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.doCancelTaskAndDescendants(TaskCancellationService.java:141)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.lambda$cancelTaskAndDescendants$0(TaskCancellationService.java:120)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ResultDeduplicator.executeOnce(ResultDeduplicator.java:49)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.cancelTaskAndDescendants(TaskCancellationService.java:117)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService$BanParentRequestHandler.messageReceived(TaskCancellationService.java:372)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService$BanParentRequestHandler.messageReceived(TaskCancellationService.java:356)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:75)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendLocalRequest(TransportService.java:1085)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$3.sendRequest(TransportService.java:160)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:982)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:877)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.setBanOnChildConnections(TaskCancellationService.java:195)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.doCancelTaskAndDescendants(TaskCancellationService.java:146)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.lambda$cancelTaskAndDescendants$0(TaskCancellationService.java:120)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ResultDeduplicator.executeOnce(ResultDeduplicator.java:49)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.cancelTaskAndDescendants(TaskCancellationService.java:117)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager.cancelTaskAndDescendants(TaskManager.java:812)
		at org.elasticsearch.xpack.esql.plugin.ComputeListener.lambda$acquireAvoid$2(ComputeListener.java:65)
		... 28 more
		Suppressed: org.elasticsearch.ElasticsearchException: 1 further exceptions were dropped
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:125)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:183)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:399)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:121)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:386)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:306)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:351)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:249)
			at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$PendingInstances.finishInstance(ExchangeSourceHandler.java:266)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.close(Releasables.java:34)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:56)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:50)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:308)
			... 21 more
üsuppressedøúBWtask_cancelled_exceptionCScancelled on failureDäorg.elasticsearch.tasks.TaskCancelledException: cancelled on failure
	at org.elasticsearch.compute.operator.Driver.cancel(Driver.java:304)
	at org.elasticsearch.compute.operator.DriverTaskRunner$DriverRequest$1.onCancelled(DriverTaskRunner.java:101)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.CancellableTask.cancel(CancellableTask.java:43)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager$CancellableTaskHolder.cancel(TaskManager.java:617)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager.cancel(TaskManager.java:274)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.doCancelTaskAndDescendants(TaskCancellationService.java:141)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.lambda$cancelTaskAndDescendants$0(TaskCancellationService.java:120)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ResultDeduplicator.executeOnce(ResultDeduplicator.java:49)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.cancelTaskAndDescendants(TaskCancellationService.java:117)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService$BanParentRequestHandler.messageReceived(TaskCancellationService.java:372)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService$BanParentRequestHandler.messageReceived(TaskCancellationService.java:356)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:75)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendLocalRequest(TransportService.java:1085)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$3.sendRequest(TransportService.java:160)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:982)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:877)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.setBanOnChildConnections(TaskCancellationService.java:195)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.doCancelTaskAndDescendants(TaskCancellationService.java:146)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.lambda$cancelTaskAndDescendants$0(TaskCancellationService.java:120)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ResultDeduplicator.executeOnce(ResultDeduplicator.java:49)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.cancelTaskAndDescendants(TaskCancellationService.java:117)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager.cancelTaskAndDescendants(TaskManager.java:812)
	at org.elasticsearch.xpack.esql.plugin.ComputeListener.lambda$acquireAvoid$2(ComputeListener.java:65)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:187)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:192)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:306)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:409)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:264)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:92)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:48)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleResponse(TransportService.java:1775)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1498)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1595)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel$1.doRun(TransportService.java:1575)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
	Suppressed: org.elasticsearch.ElasticsearchException: 1 further exceptions were dropped
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:125)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:183)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:399)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:121)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:386)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:306)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:351)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:249)
		at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$PendingInstances.finishInstance(ExchangeSourceHandler.java:266)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.close(Releasables.java:34)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:56)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:50)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:308)
		... 21 more
üEøúBHexceptionC`1 further exceptions were droppedDäorg.elasticsearch.ElasticsearchException: 1 further exceptions were dropped
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:125)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:183)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:399)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:121)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:386)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:306)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:351)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:249)
	at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$PendingInstances.finishInstance(ExchangeSourceHandler.java:266)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.close(Releasables.java:34)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:56)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:50)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:308)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:409)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:264)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:92)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:48)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleResponse(TransportService.java:1775)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1498)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1595)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel$1.doRun(TransportService.java:1575)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
üûùûùûùBHexceptionC!Däorg.elasticsearch.ElasticsearchException
	at org.elasticsearch.compute.operator.FailureCollector.unwrapAndCollect(FailureCollector.java:51)
	at org.elasticsearch.xpack.esql.plugin.ComputeListener.lambda$acquireAvoid$2(ComputeListener.java:61)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:187)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:192)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:306)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:409)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:264)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:92)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:48)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleResponse(TransportService.java:1775)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1498)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1595)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel$1.doRun(TransportService.java:1575)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
	Suppressed: org.elasticsearch.tasks.TaskCancelledException: cancelled on failure
		at org.elasticsearch.compute.operator.Driver.cancel(Driver.java:304)
		at org.elasticsearch.compute.operator.DriverTaskRunner$DriverRequest$1.onCancelled(DriverTaskRunner.java:101)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.CancellableTask.cancel(CancellableTask.java:43)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager$CancellableTaskHolder.cancel(TaskManager.java:617)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager.cancel(TaskManager.java:274)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.doCancelTaskAndDescendants(TaskCancellationService.java:141)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.lambda$cancelTaskAndDescendants$0(TaskCancellationService.java:120)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ResultDeduplicator.executeOnce(ResultDeduplicator.java:49)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.cancelTaskAndDescendants(TaskCancellationService.java:117)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService$BanParentRequestHandler.messageReceived(TaskCancellationService.java:372)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService$BanParentRequestHandler.messageReceived(TaskCancellationService.java:356)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:75)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendLocalRequest(TransportService.java:1085)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$3.sendRequest(TransportService.java:160)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:982)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:877)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.setBanOnChildConnections(TaskCancellationService.java:195)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.doCancelTaskAndDescendants(TaskCancellationService.java:146)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.lambda$cancelTaskAndDescendants$0(TaskCancellationService.java:120)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ResultDeduplicator.executeOnce(ResultDeduplicator.java:49)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.cancelTaskAndDescendants(TaskCancellationService.java:117)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager.cancelTaskAndDescendants(TaskManager.java:812)
		at org.elasticsearch.xpack.esql.plugin.ComputeListener.lambda$acquireAvoid$2(ComputeListener.java:65)
		... 28 more
		Suppressed: org.elasticsearch.ElasticsearchException: 1 further exceptions were dropped
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:125)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:183)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:399)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:121)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:386)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:306)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:351)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:249)
			at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$PendingInstances.finishInstance(ExchangeSourceHandler.java:266)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.close(Releasables.java:34)
			at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:56)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:50)
			at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:308)
			... 21 more
üEøúBWtask_cancelled_exceptionCScancelled on failureDäorg.elasticsearch.tasks.TaskCancelledException: cancelled on failure
	at org.elasticsearch.compute.operator.Driver.cancel(Driver.java:304)
	at org.elasticsearch.compute.operator.DriverTaskRunner$DriverRequest$1.onCancelled(DriverTaskRunner.java:101)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.CancellableTask.cancel(CancellableTask.java:43)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager$CancellableTaskHolder.cancel(TaskManager.java:617)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager.cancel(TaskManager.java:274)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.doCancelTaskAndDescendants(TaskCancellationService.java:141)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.lambda$cancelTaskAndDescendants$0(TaskCancellationService.java:120)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ResultDeduplicator.executeOnce(ResultDeduplicator.java:49)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.cancelTaskAndDescendants(TaskCancellationService.java:117)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService$BanParentRequestHandler.messageReceived(TaskCancellationService.java:372)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService$BanParentRequestHandler.messageReceived(TaskCancellationService.java:356)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:75)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendLocalRequest(TransportService.java:1085)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$3.sendRequest(TransportService.java:160)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:982)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:877)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.setBanOnChildConnections(TaskCancellationService.java:195)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.doCancelTaskAndDescendants(TaskCancellationService.java:146)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.lambda$cancelTaskAndDescendants$0(TaskCancellationService.java:120)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ResultDeduplicator.executeOnce(ResultDeduplicator.java:49)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskCancellationService.cancelTaskAndDescendants(TaskCancellationService.java:117)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.tasks.TaskManager.cancelTaskAndDescendants(TaskManager.java:812)
	at org.elasticsearch.xpack.esql.plugin.ComputeListener.lambda$acquireAvoid$2(ComputeListener.java:65)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:187)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:192)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:306)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:409)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:264)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:92)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:48)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleResponse(TransportService.java:1775)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1498)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1595)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel$1.doRun(TransportService.java:1575)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
	Suppressed: org.elasticsearch.ElasticsearchException: 1 further exceptions were dropped
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:125)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:183)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:399)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:121)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:386)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:306)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:351)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:249)
		at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$PendingInstances.finishInstance(ExchangeSourceHandler.java:266)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.close(Releasables.java:34)
		at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:56)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:50)
		at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:308)
		... 21 more
üEøúBHexceptionC`1 further exceptions were droppedDäorg.elasticsearch.ElasticsearchException: 1 further exceptions were dropped
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:125)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onResponse(ActionListenerImplementations.java:183)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:399)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:121)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:386)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:306)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:351)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:249)
	at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$PendingInstances.finishInstance(ExchangeSourceHandler.java:266)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.close(Releasables.java:34)
	at org.elasticsearch.base@8.16.0/org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:56)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:50)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:308)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:409)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:264)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:63)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:74)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:92)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:48)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleResponse(TransportService.java:1775)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1498)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1595)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.transport.TransportService$DirectResponseChannel$1.doRun(TransportService.java:1575)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.16.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
üûùûùûstatus$¨û
```

**Issue Reasons:**
- [main] 2 consecutive failures in step release-tests
- [main] 2 failures in test testTopLevelFilterBoolMerged {SYNC} (1.2% fail rate in 161 executions)
- [main] 2 failures in step release-tests (40.0% fail rate in 5 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

```
Caused by: java.lang.AssertionError: java.lang.ClassCastException: class org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput cannot be cast to class org.elasticsearch.xpack.esql.io.stream.PlanStreamInput (org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput is in module org.elasticsearch.server@8.16.0 of loader 'app'; org.elasticsearch.xpack.esql.io.stream.PlanStreamInput is in unnamed module of loader java.net.FactoryURLClassLoader @1584c019)
	at org.elasticsearch.transport.InboundHandler.handleRequest(InboundHandler.java:262) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:116) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:97) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:821) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:124) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:96) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:61) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.netty4.Netty4MessageInboundHandler.channelRead(Netty4MessageInboundHandler.java:57) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[?:?]
	... 18 more
Caused by: java.lang.ClassCastException: class org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput cannot be cast to class org.elasticsearch.xpack.esql.io.stream.PlanStreamInput (org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput is in module org.elasticsearch.server@8.16.0 of loader 'app'; org.elasticsearch.xpack.esql.io.stream.PlanStreamInput is in unnamed module of loader java.net.FactoryURLClassLoader @1584c019)
	at org.elasticsearch.xpack.esql.querydsl.query.SingleValueQuery$Builder.<init>(SingleValueQuery.java:110) ~[?:?]
	at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteableCollectionAsList(NamedWriteableAwareStreamInput.java:45) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.index.query.AbstractQueryBuilder.readQueries(AbstractQueryBuilder.java:275) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.index.query.BoolQueryBuilder.<init>(BoolQueryBuilder.java:74) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:56) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:32) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.search.builder.SubSearchSourceBuilder.<init>(SubSearchSourceBuilder.java:79) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.common.io.stream.StreamInput.readCollection(StreamInput.java:1290) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.common.io.stream.StreamInput.readCollectionAsList(StreamInput.java:1162) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.search.builder.SearchSourceBuilder.<init>(SearchSourceBuilder.java:239) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.common.io.stream.StreamInput.readOptionalWriteable(StreamInput.java:1088) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.action.search.CanMatchNodeRequest.<init>(CanMatchNodeRequest.java:185) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.RequestHandlerRegistry.newRequest(RequestHandlerRegistry.java:62) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundHandler.handleRequest(InboundHandler.java:260) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:116) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:97) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:821) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:124) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:96) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:61) ~[elasticsearch-8.16.0.jar:?]
	at org.elasticsearch.transport.netty4.Netty4MessageInboundHandler.channelRead(Netty4MessageInboundHandler.java:57) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[?:?]
	... 18 more
```

--- a/docs/changelog/111779.yaml
+++ b/docs/changelog/111779.yaml
@@ -1,0 +1,7 @@
+pr: 111779
+summary: "ESQL: Fix serialization during `can_match`"
+area: ES|QL
+type: bug
+issues:
+ - 111701
+ - 111726

--- a/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/tree/Source.java
+++ b/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/tree/Source.java
@@ -51,6 +51,21 @@ public final class Source implements Writeable {
         return new Source(new Location(line, charPositionInLine), text);
     }
 
+    /**
+     * Read the components of a {@link Source} and throw it away, returning
+     * {@link Source#EMPTY}. Use this when you will never use the {@link Source}
+     * and there is no chance of getting a {@link PlanStreamInput}.
+     */
+    public static Source readEmpty(StreamInput in) throws IOException {
+        if (in.readBoolean() == false) {
+            return EMPTY;
+        }
+        in.readInt();
+        in.readInt();
+        in.readInt();
+        return Source.EMPTY;
+    }
+
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         if (this == EMPTY) {

--- a/x-pack/plugin/esql-core/src/test/java/org/elasticsearch/xpack/esql/core/tree/SourceTests.java
+++ b/x-pack/plugin/esql-core/src/test/java/org/elasticsearch/xpack/esql/core/tree/SourceTests.java
@@ -6,13 +6,17 @@
  */
 package org.elasticsearch.xpack.esql.core.tree;
 
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.test.ESTestCase;
 
+import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
 import java.util.function.Function;
 
 import static org.elasticsearch.test.EqualsHashCodeTestUtils.checkEqualsAndHashCode;
+import static org.hamcrest.Matchers.equalTo;
 
 public class SourceTests extends ESTestCase {
     public static Source randomSource() {
@@ -41,5 +45,17 @@ public class SourceTests extends ESTestCase {
             l -> new Source(l.source().getLineNumber(), l.source().getColumnNumber() - 1, l.text()),
             SourceTests::mutate
         );
+    }
+
+    public void testReadEmpty() throws IOException {
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            randomSource().writeTo(out);
+            int test = randomInt();
+            out.writeInt(test);
+            try (StreamInput in = out.bytes().streamInput()) {
+                assertThat(Source.readEmpty(in), equalTo(Source.EMPTY));
+                assertThat(in.readInt(), equalTo(test));
+            }
+        }
     }
 }

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQuery.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQuery.java
@@ -107,7 +107,15 @@ public class SingleValueQuery extends Query {
             this.next = in.readNamedWriteable(QueryBuilder.class);
             this.field = in.readString();
             if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_SINGLE_VALUE_QUERY_SOURCE)) {
-                this.source = Source.readFrom((PlanStreamInput) in);
+                if (in instanceof PlanStreamInput psi) {
+                    this.source = Source.readFrom(psi);
+                } else {
+                    /*
+                     * For things like CanMatch we serialize without the Source. But we
+                     * don't use it, so that's ok.
+                     */
+                    this.source = Source.readEmpty(in);
+                }
             } else if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_12_0)) {
                 this.source = readOldSource(in);
             } else {

